src/js/components/artifacts/dialogs/addartifact.js

```diff
@@ -12,42 +12,46 @@ const steps = [
   { title: 'Artifact information', component: ArtifactInformationForm }
 ];
 
-export class AddArtifactDialog extends React.Component {
-  constructor(props, context) {
-    super(props, context);
-    this.state = {
-      activeStep: 0,
-      customDeviceTypes: '',
-      destination: !props.onboardingState.complete ? '/var/www/localhost/htdocs' : '',
-      file: null,
-      selectedDeviceTypes: []
-    };
-  }
+export const AddArtifactDialog = ({
```

Comment:
I'm happy to see we are moving to functional components!

Reply:
we're down to less than 20 class based components by now - unfortunately another small batch is waiting in the monitoring feature branch - but soon 🎉 (and related: #1803)